src/graphics/program-lib/programs/standard-node.js

```diff
@@ -0,0 +1,93 @@
+import { standard } from './standard.js';
+
+// just do a shallow copy standard
+var standardnode = Object.assign({}, standard);
+
+// override a couple of functions - the overridden functions have been altered in standard to accept extra params
+standardnode._generateKey = standard.generateKey;
+standardnode.generateKey = function (options) {
+    return this._generateKey(options, options._shaderGraphChunk.id);
+};
+
+standardnode._createShaderDefinition = standard.createShaderDefinition;
+standardnode.createShaderDefinition = function (device, options) {
+    var rootShaderGraph = null;
+    var rootDeclGLSL = '';
+    var rootCallGLSL = '';
+
+    if (options._shaderGraphChunk) {
+        rootShaderGraph = options._shaderGraphChunk;
+
+        rootDeclGLSL = rootShaderGraph.generateRootDeclGlsl();
+        rootCallGLSL = rootShaderGraph.generateRootCallGlsl();
+    }
+
+    var graphCode = [];
+
+    if (options._shaderGraphChunk) {
+        graphCode[0] = '';
+        if (rootShaderGraph.getIoPortByName('OUT_vertOff')) {
```

Comment:
also this line is called again bellow, perhaps cache result, or combine those blocks?

Reply:
I have improved this - I haven't turned it into a generic data driven system because it is still tied to the non-data driven standard material.